source/MaterialXGenOgsXml/OgsXmlGenerator.cpp

```diff
@@ -0,0 +1,230 @@
+//
+// TM & (c) 2017 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#include <MaterialXGenOgsXml/OgsXmlGenerator.h>
+#include <MaterialXGenOgsXml/GlslFragmentGenerator.h>
+
+#include <MaterialXGenShader/Shader.h>
+
+#include <MaterialXFormat/XmlIo.h>
+#include <MaterialXFormat/PugiXML/pugixml.hpp>
+
+using namespace pugi;
+
+namespace MaterialX
+{
+
+namespace
+{
+    // Data types used by OGS
+    static const std::unordered_map<const TypeDesc*, const pugi::char_t*> OGS_TYPE_MAP =
+    {
+        { Type::BOOLEAN, "bool" },
+        { Type::FLOAT, "float" },
+        { Type::INTEGER, "int" },
+        { Type::STRING, "int" },
+        { Type::COLOR2, "float2" },
+        { Type::COLOR3, "float3" },
+        { Type::COLOR4, "float4" },
+        { Type::VECTOR2, "float2" },
+        { Type::VECTOR3, "float3" },
+        { Type::VECTOR4, "float4" },
+        { Type::MATRIX33, "float4x4" },
+        { Type::MATRIX44, "float4x4" }
+    };
+
+    // Semantics used by OGS
+    static const std::unordered_map<string, const pugi::char_t*> OGS_SEMANTICS_MAP =
+    {
+        { "positionWorld", "Pw"},
+        { "positionObject", "Pm"},
+        { "normalWorld", "Nw" },
+        { "normalObject", "Nm" },
+        { "tangentWorld", "mayaTangentIn" },
+        { "bitangentWorld", "mayaBitangentIn" },
+        { "texcoord_0", "mayaUvCoordSemantic" },
+        { "color_0", "colorset" },
+
+        { "u_worldMatrix", "World" },
+        { "u_worldInverseMatrix", "WorldInverse" },
+        { "u_worldTransposeMatrix", "WorldTranspose" },
+        { "u_worldInverseTransposeMatrix", "WorldInverseTranspose" },
+
+        { "u_viewMatrix", "View" },
+        { "u_viewInverseMatrix", "ViewInverse" },
+        { "u_viewTransposeMatrix", "ViewTranspose" },
+        { "u_viewInverseTransposeMatrix", "ViewInverseTranspose" },
+
+        { "u_projectionMatrix", "Projection" },
+        { "u_projectionInverseMatrix", "ProjectionInverse" },
+        { "u_projectionTransposeMatrix", "ProjectionTranspose" },
+        { "u_projectionInverseTransposeMatrix", "ProjectionInverseTranspose" },
+
+        { "u_worldViewMatrix", "WorldView" },
+        { "u_viewProjectionMatrix", "ViewProjection" },
+        { "u_worldViewProjectionMatrix", "WorldViewProjection" },
+
+        { "u_viewDirection", "ViewDirection" },
+        { "u_viewPosition", "WorldCameraPosition" },
```

Comment:
I've tested this out and the XML will produce the correctly tracked camera position when Maya binds the shader and as you found out before the incorrect value when using the glslPlugin (where ViewPosition is required).